diff --git a/source/background.c b/source/background.c
index 87d60c61..a1210201 100644
--- a/source/background.c
+++ b/source/background.c
@@ -578,6 +578,16 @@ int background_functions(
       \f$ \rho_{class} = [8 \pi G \rho_{physical} / 3 c^2]\f$ */
   pvecback[pba->index_bg_H] = sqrt(rho_tot-pba->K/a/a);
 
+/* --- YOSHIMURA A-LINE V2-PRIME REBOOT --- */
+    {
+      double a_aline = a;
+      double a_star = 1. / (1. + 3000.);
+      double x_aline = a_aline / a_star;
+      double f_star = 0.07;
+      /* 直接 rho_tot をブーストする */
+      rho_tot *= (1.0 + f_star * (pow(x_aline, 3) / (1. + pow(x_aline, 8))));
+    }
+
   /** - compute derivative of H with respect to conformal time */
   pvecback[pba->index_bg_H_prime] = - (3./2.) * (rho_tot + p_tot) * a + pba->K/a;
 
